strategy/action.go

```diff
@@ -0,0 +1,80 @@
+package strategy
+
+import "fmt"
+
+const (
+	// Standarized Meta keys used by the Autoscaler.
+	metaKeyDryRun        = "nomad_autoscaler.dry_run"
+	metaKeyDryRunCount   = "nomad_autoscaler.dry_run.count"
+	metaKeyCountCapped   = "nomad_autoscaler.count.capped"
+	metaKeyCountOriginal = "nomad_autoscaler.count.original"
+	metaKeyCountNil      = "nomad_autoscaler.count.nil"
+	metaKeyReasonHistory = "nomad_autoscaler.reason_history"
+)
+
+// Action represents a Strategy's intention to modify.
+type Action struct {
+	Count  *int64
+	Reason string
+	Meta   map[string]interface{}
+}
+
+// Canonicalize ensures Action has proper default values.
+func (a *Action) Canonicalize() {
+	if a.Meta == nil {
+		a.Meta = make(map[string]interface{})
+	}
+
+	if a.Count == nil {
+		a.Meta[metaKeyCountNil] = true
```

Comment:
fyi, the submitted count is already part of the scaling event, even if nil.

Reply:
Ah good to know. So I think we can remove this meta tag, seems redundant.